apps/mobile/src/utils/formatters.ts

```diff
@@ -47,3 +47,42 @@ const calculateRatio = (a: Quantity, b: Quantity) => {
   }
   return asDecimal(BigInt(a.amount), a.decimals) / asDecimal(BigInt(b.amount), b.decimals)
 }
+
+export const customIntl = (number = 0, locale = 'en-US', currency = 'USD') => {
+  const absNumber = Math.abs(number)
+  const units = ['', 'K', 'M', 'B', 'T']
+  let unitIndex = 0
+  let num = absNumber
+
+  while (num >= 1000 && unitIndex < units.length - 1) {
+    num /= 1000
+    unitIndex++
+  }
+
+  const formattedNumber = num.toLocaleString(locale, {
+    minimumFractionDigits: 0,
+    maximumFractionDigits: 2,
+  })
+
+  const formattedCurrency = new Intl.NumberFormat(locale, {
+    style: 'currency',
+    currency: currency,
+    minimumFractionDigits: 0,
+    maximumFractionDigits: 2,
+  }).format(Number(formattedNumber))
+
+  return `${formattedCurrency}${units[unitIndex]}`
```

Comment:
Why do you need this custom unit handling? The `compact` option of NumberFormat does exactly that. That's what the web app is using. Please don't reinvent the wheel.

<img width="549" alt="Screenshot 2025-03-13 at 15 48 29" src="https://github.com/user-attachments/assets/a6fa8218-4872-4f78-ad4a-db3d0817f80a" />

Reply:
@katspaugh there is a bug in Hermes and the compact mode doesn't work on iOS. Jonathan and I discussed this and went this route: https://github.com/facebook/hermes/issues/1035